lib/charms/oauth2_proxy_k8s/v0/auth_proxy.py

```diff
@@ -371,7 +371,7 @@ def get_app_names(self) -> List[str]:
         app_names = []
         for relation in self._charm.model.relations[self._relation_name]:
             if relation.data[relation.app]:
-                app_names.append(relation.app.name)
+                app_names.append(relation.data[relation.app]["app_name"])
```

Comment:
are we able to fetch also the model?

if we do this, can this potentially solve the issue of same-named apps in different models that we were discussing yesterday?